sympy/printing/pycode.py

```diff
@@ -748,3 +748,110 @@ def _print_Function(self, expr):
 
     def _print_Pow(self, expr, rational=False):
         return self._hprint_Pow(expr, rational=rational, sqrt='sympy.sqrt')
+
+
+_not_in_arb = ['besseli', 'besselj', 'besselk', 'bessely', 'beta',
+               'factorial', 'frac', 'fresnelc', 'fresnels', 'hyper',
+               'hypot', 'log10', 'loggamma', 'meijerg', 'sign']
+
+_in_arb = [(k, v) for k, v in _known_functions_mpmath.items() if k not in _not_in_arb]
+
+_known_functions_arb = dict(_in_arb, **{
+    'sign': 'sgn',
+    'factorial': 'fac',
+    'fresnelc': 'fresnel_c',
+    'fresnels': 'fresnel_s',
+    'hyper': 'hypgeom',
+})
+
+_known_constants_arb = {
+    'Exp1': 'const_e',
+    'Pi': 'pi',
+    'EulerGamma': 'const_euler',
+    'Catalan': 'const_catalan',
+    'NaN': 'nan',
+    'Infinity': 'pos_inf',
+    'NegativeInfinity': 'neg_inf'
+}
+
+
+class ArbPrinter(PythonCodePrinter):
+    """
+    Code printer for arb (arbitrary precision ball arithmetic).
+    """
+    printmethod = "_arbcode"
+
+    language = "Python with flint.arb"
+
+    _kf = dict(chain(
+        _known_functions.items(),
+        [(k, 'arb.' + v) for k, v in _known_functions_arb.items()]
+    ))
+    _kc = {k: 'arb.'+v for k, v in _known_constants_arb.items()}
+
+    def _print_Integer(self, arg):
+        return "arb(%d)" % arg
+
+    def _print_Float(self, arg):
+        return 'arb("%s")' % str(arg)
+
+    def _bessel_x(self, x, b):
+        return "{}.bessel_{}({})".format(self._print(b.args[1]), x, self._print(b.args[0]))
```

Comment:
This same sort of fix is actually needed for most arb functions. Basically everything is a method and so it should be printed as a method call.

Reply:
yes, that sounds reasonable. Right now I don't have time to extend this beyond the things I happen to need myself.

Reply:
feel free to push directly into this branch if you feel like doing so!